src/core/CellStructure.hpp

```diff
@@ -33,6 +33,7 @@
 #include "algorithm/link_cell.hpp"
 #include "bond_error.hpp"
 #include "ghosts.hpp"
+#include "nonbonded_interactions/nonbonded_interaction_data.hpp"
```

Comment:
This is not needed (you ended up not using the constant). It also drags in globals, so should not be here in any case.

Reply:
I need this include for the prototype of `maximal_cutoff()`

Reply:
But that operates on globals?

Reply:
So please don't do this. This is slightly shocking to me.

Reply:
The function iterates over all interactions data structures. I could do a forward declaration of the `maximal_cutoff()` prototype to avoid including the complete header file.

Reply:
This is worse. If you need to know the maximal cutoff, it should be passed in right? You should also check if you can move the decision if the function should be called up the call stack.